### ISSUE TITLE
WCAG: Skip link targets empty anchor element (Issue A1)

### ISSUE THREAD

**Issue by CJE001** [civictheme/uikit#355]
### Summary

Via Vision Australia assessment: August 2024

Note: DTA have a 90 day remediation period to address the identified issues within the audit, all issues must be resolved to obtain WCAG 2.2 certification for digital.gov.au

### Steps to reproduce

Global issue – Skip to main content link
[digital.gov.au beta | digital.gov.au](https://www.digital.gov.au/) (https://www.digital.gov.au)

The global skip link used in the site header directs to an empty < a > (anchor) element. Additionally, there are two empty anchor elements in the code with duplicate id attributes. The skip link is directing to the first empty anchor element.

![Picture5](https://github.com/user-attachments/assets/9b02768a-bfb1-43b0-9ae0-29fab4b06e9a)

### Observed outcome

```
<div class="ct-skip-link ct-theme-light ">
<a class="ct-link ct-theme-light ct-skip-link__link ct-visually-hidden ct-focusable" href="#main-content" title="Skip to main content">
Skip to main content
</a>
</div>
… 
<a id="main-content" tabindex="-1"></a>
<main class="ct-layout ct-theme-light ct-vertical-spacing--top " role="main">
<a id="main-content" tabindex="-1"></a>
… 
</main>
```
### Why this matters

As the target for the skip link is empty, has a generic role and no accessible name, screen readers may announce 'blank' or nothing at all when the element receives keyboard focus. Users may struggle to determine if activating the link was successful, or where they are on the page. 

### Expected outcome

Remove the empty anchor elements from the code. Set the target of the skip link to the <main> element, by adding an id attribute that matches the target of the skip link.

The element should also be given a tabindex="-1" to ensure that the skip link works for users across various operating systems. 

Code example

```
<main id="main-content" tabindex="-1" class="ct-layout ct-theme-light ct-vertical-spacing--top " role="main">…
</main>
```

**Comment by fionamorrison23** [comment]
Hi @CJE001  would you be able to provide the priority here as you have done on other tickets? Thanks.

**Comment by daniel-grynenko** [comment]
Test Env: DEV
Test Status: PASS
Test Result: 
1. Verified the skip link correctly moves focus to the main content instead of an empty anchor.
2. Confirmed id="main-content" is applied to the banner if available. Where no banner is present, it will appear in <main>
3. Validated no duplicate id="main-content" elements exist.
4. Tested keyboard navigation to confirm pressing the Enter key moves focus past the navigation.
5. Verified screen reader behaviour correctly announces and directs users to meaningful content.
6. Confirmed removal of old empty anchor elements does not impact functionality or accessibility.

<img width="1507" alt="Image" src="https://github.com/user-attachments/assets/10d8c845-add2-41fd-8987-695124f1ac62" />
